src/main/java/io/aiven/kafka/connect/http/sender/AbstractHttpSender.java

```diff
@@ -48,12 +48,25 @@ protected AbstractHttpSender(
     }
 
     public final HttpResponse<String> send(final String body) {
-        final var requestBuilder =
-                httpRequestBuilder.build(config).POST(HttpRequest.BodyPublishers.ofString(body));
+        final var requestBuilder = prepareRequest(body);
         return sendWithRetries(requestBuilder, HttpResponseHandler.ON_HTTP_ERROR_RESPONSE_HANDLER,
                 config.maxRetries());
```

Comment:
We can improve the naming here
```suggestion
        final var request = prepareRequest(body);
        return sendWithRetries(request, HttpResponseHandler.ON_HTTP_ERROR_RESPONSE_HANDLER, config.maxRetries());
```